opcua/server/history.py

```diff
@@ -108,20 +117,191 @@ def read_node_history(self, node, start, end, nb_values):
+                                                                                                 datavalue.Value.VariantType.value))
+        except sqlite3.Error as e:
+            print(node_id, 'Historizing SQL Insert Error:', e)
+
+        # get this node's period from the period dict and calculate the limit
+        period = self._datachanges_period[node_id]
+        date_limit = datetime.now() - period
+
+        # after the insert, delete all values older than period
+        try:
+            self._c_sub.execute('DELETE FROM "{tn}" WHERE ServerTimestamp < ?'.format(tn=node_id), (date_limit.isoformat(' '),))
+        except sqlite3.Error as e:
+            print(node_id, 'Historizing SQL Delete Old Data Error:', e)
+
+        self._conn_sub.commit()
+
+    def read_node_history(self, node, start, end, nb_values):
+        if self._conn_read is None:
+            self._conn_read = sqlite3.connect(self._db_file, detect_types=sqlite3.PARSE_DECLTYPES)
+            self._c_read = self._conn_read.cursor()
+
+        if end is None:
+            end = datetime.now() + timedelta(days=1)
+        if start is None:
+            start = ua.DateTimeMinValue
+
+        node_id = str(node.NodeId.NamespaceIndex) + '_' + str(node.NodeId.Identifier)
+        cont = None
+        results = []
+
+        start_time = start.isoformat(' ')
+        end_time = end.isoformat(' ')
+
+        # select values from the database
+        try:
+            for row in self._c_read.execute('SELECT * FROM "{tn}" WHERE "ServerTimestamp" BETWEEN ? AND ? '
+                                           'LIMIT ?'.format(tn=node_id), (start_time, end_time, nb_values,)):
+
+                dv = ua.DataValue(ua.Variant(row[3], self._get_variant_type(row[4])))
+                dv.ServerTimestamp = row[0]
+                dv.SourceTimestamp = row[1]
+                dv.StatusCode = ua.StatusCode(row[2])
+
+                results.append(dv)
+
+        except sqlite3.Error as e:
+            print(node_id, 'Historizing SQL Read Error:', e)
+
+        return results, cont
+
+    def new_historized_event(self, event, period):
+        raise NotImplementedError
+
+    def save_event(self, event):
+        raise NotImplementedError
+
+    def read_event_history(self, start, end, evfilter):
+        raise NotImplementedError
+
+    # convert the node UA Variant type to an SQL supported type
+    # FIXME: this could lead to lost precision! Better way to store the data? Add custom types to SQL?
+    def _get_sql_type(self, node):
+        node_type = node.get_data_type()
+        # see object_ids.py
+        if node_type.Identifier in (10, 11,):  # Float, Double
+            return 'REAL'
+        elif node_type.Identifier in (4, 5, 6, 7, 8, 9,):  # Ints
+            return "INT"
+        elif node_type.Identifier in (12,):  # String
+            return "TEXT"
+        else:
+            return "NULL"
```

Comment:
use the VariantType attribute from variant, data_type form Node is redundant and can be wrong